lib/cli/transaction.rs

```diff
@@ -90,6 +90,24 @@ pub fn make_transaction(
     crate::output_transaction(output, &transaction).map_err(CliError::from)
 }
 
+/// Reads a previously-saved [`TransactionV1`] from a file, cryptographically signs it, and outputs it to a
+/// file or stdout.
+///
+/// `maybe_output_path` specifies the output file path, or if empty, will print it to `stdout`.  If
+/// `force` is true, and a file exists at `maybe_output_path`, it will be overwritten.  If `force`
+/// is false and a file exists at `maybe_output_path`, [`Error::FileAlreadyExists`] is returned
+/// and the file will not be written.
+pub fn sign_transaction_file(
+    input_path: &str,
+    secret_key_path: &str,
+    maybe_output_path: &str,
```

Comment:
You don't need to specify maybe_xxx or make it an Option in this case, relying on an empty string does not read very well on consumer point of view 

casper_client::cli::sign_transaction_file(input_path, secret_key, "", true). 

It's also a bit strange to have a mandatory input path and a non mandatory output path but I guess this will be okay for now.

Reply:
I agree that relying on a empty string isn't ideal. The parameter should be optional as the default behavior of the client when no output path is specified is to output to stdout. I would argue that, as optional parameter, it should retain the name `maybe_output_path`. However, when I went to change this to an `Option<&str> along with changing the parsing behavior, I realized that functions related to deploys also depend on the same parsing function. Therefore, I would need to change the API of previously exposed library functions, or write specific parsing code for `OutputKind` that is unique to transactions. The client should, In my opinion, maintain consistent behavior between the two.

Reply:
Of course I can live with current code but I would start wrapping that maybe pattern where ever I can in the API, if I undesrtand correctly user is not using  crate::sign_transaction_file directly but this interface so I would mask this little empty string trick

I suggest thus, but not a perquisite 

```
pub fn sign_transaction_file(
    input_path: &str,
    secret_key_path: &str,
    maybe_output_path: Option<&str>,
    force: bool,
) -> Result<(), CliError> {
    let output = parse::output_kind(maybe_output_path.unwrap_or(""), force);
    let secret_key = parse::secret_key_from_file(secret_key_path)?;
    crate::sign_transaction_file(input_path, &secret_key, output).map_err(CliError::from)
}
```

Cheers

Reply:
I ended up going with this suggestion as it's a good interim solution. I'll see if it would be worth creating a ticket to make this uniform across the client.